EmailSender.Application/Services/MailService.cs

```diff
@@ -6,26 +6,40 @@
 
 namespace EmailSender.Application.Services
 {
-    public class MailService
+    public class MailService : IEmailSender
     {
         private readonly SendGridOptions _mailOptions;
-        private readonly SendGridClient _client;
 
         public MailService(IOptions<SendGridOptions> mailOptions)
         {
             _mailOptions = mailOptions.Value;
-            _client = new SendGridClient(_mailOptions.SendGridAPIKey);
         }
 
-        public async Task SendEmail(MailModel model)
+        public async Task SendEmailAsync(MailModel mailModel)
         {
-            var mail = new SendGridMessage
+            if (string.IsNullOrEmpty(_mailOptions.SendGridAPIKey))
```

Comment:
Usually you will see an error that options are not fulfilled so we can remove this validation